presto-spi/src/main/java/com/facebook/presto/spi/spiller/Spiller.java

```diff
@@ -0,0 +1,31 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.spi.spiller;
+
+import com.facebook.presto.spi.Page;
+
+import java.io.Closeable;
+import java.util.Iterator;
+import java.util.List;
+import java.util.concurrent.CompletableFuture;
+
+public interface Spiller extends Closeable
+{
+    public CompletableFuture<?> spill(Iterator<Page> pageIterator);
+
+    public List<Iterator<Page>> getSpills();
```

Comment:
public dropped.

Comments added.

@kokosing `getSpills` doesn't merge the spilled data, it just provides List of previously spilled streams. One stream is a collection of `Pages` hence return type of `List<Iterator<Page>>`.
```

@kokosing `getSpills` doesn't merge the spilled data, it just provides List of previously spilled streams. One stream is a collection of `Pages` hence return type of `List<Iterator<Page>>`.